### PR TITLE
Explicitly document getUserMedia() constraints shape

### LINKED ISSUE #22962: Missing description of getUserMedia constraints.
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia

### What specific section or headline is this issue about?

Syntax

### What information was incorrect, unhelpful, or incomplete?

The `constraints` parameter is not described except through some examples. In particular, I have no idea what values are valid for the "audio" attribute except for "true" that I saw in the example.

From other sources I know I can use "echoCancellation" but I think that has some options and I have no idea what they are or how to know if they are supported.

### What did you expect to see?

I'd like the word "constraints" to be a link to a page that describes all of the possible options. Note - this wouldn't be as important if I could find that information somewhere else but I've been searching for it and don't see it.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

My particular issue is that I'm trying to make this work on iPhone 14 and I'd like to know what values are valid for that device.

Since I'm trying to get this to work in as many devices as possible, I'd like to know what the device and browser differences are so I wish there were a resource that listed what options work on what devices.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/mediadevices/getusermedia`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/mediadevices/getusermedia/index.md
* Last commit: https://github.com/mdn/content/commit/af967bb052695baa631c004b9b149c6f91920776
* Document last modified: 2022-12-12T20:23:34.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
### Description

Fix https://github.com/mdn/content/issues/22962

### CODE REVIEW

File: files/en-us/web/api/mediadevices/getusermedia/index.md [hunk 1]
```diff
@@ -361,6 +210,163 @@ situations in which `getUserMedia()` is not permitted to be called:
...
+```
+
+If no camera exists with this resolution or higher, then the returned promise will
+be rejected with `OverconstrainedError`, and the user will not be
+prompted.
+
+The reason for the difference in behavior is that the keywords `min`,
+`max`, and `exact` are inherently mandatory. Whereas plain
+values and a keyword called `ideal` are not. Here's a full example:
+
```
Comment (sideshowbarker) [inline files/en-us/web/api/mediadevices/getusermedia/index.md:295]:
```suggestion
The reason for the difference in behavior is that the keywords `min`,
`max`, and `exact` are inherently mandatory — whereas plain
values and a keyword called `ideal` are not. Here's a full example:

```

File: files/en-us/web/api/mediadevices/getusermedia/index.md [hunk 2]
```diff
@@ -361,6 +210,163 @@ situations in which `getUserMedia()` is not permitted to be called:
...
+  },
+});
+```
+
+An `ideal` value, when used, has gravity, which means that the browser
+will try to find the setting (and camera, if you have more than one), with the
+smallest [fitness distance](https://w3c.github.io/mediacapture-main/#dfn-fitness-distance) from the ideal values given.
+
+Plain values are inherently ideal, which means that the first of our resolution
+examples above could have been written like this:
```
Comment (sideshowbarker) [inline files/en-us/web/api/mediadevices/getusermedia/index.md:311]:
```suggestion
An `ideal` value, when used, has gravity — which means that the browser
will try to find the setting (and camera, if you have more than one), with the
smallest [fitness distance](https://w3c.github.io/mediacapture-main/#dfn-fitness-distance) from the ideal values given.

Plain values are inherently ideal, which means that the first of our resolution
examples above could have been written like this:
```

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/api/mediadevices/getusermedia/index.md [hunk 3]
```diff
@@ -361,6 +210,163 @@ situations in which `getUserMedia()` is not permitted to be called:
...
+  },
+});
+```
+
+If no camera exists with this resolution or higher, then the returned promise will
+be rejected with `OverconstrainedError`, and the user will not be
+prompted.
+
+The reason for the difference in behavior is that the keywords `min`,
+`max`, and `exact` are inherently mandatory — whereas plain
```
Comment (sideshowbarker) [inline files/en-us/web/api/mediadevices/getusermedia/index.md]:
```suggestion
`max`, and `exact` are inherently mandatory — whereas plain
```

**Comment by Josh-Cena** [comment]
On a mac, <kbd>Option</kbd>+<kbd>Space</kbd> produces an NBSP, so it's super easy to sneak them in when typing <kbd>Option</kbd>+<kbd>Shift</kbd>+<kbd>-</kbd>...

**Comment by Josh-Cena** [comment]
I changed my muscle memory so my thumb does not linger on <kbd>Option</kbd> too long. Alternatively, maybe we can get into the habit of not putting spaces around em-dashes—it looks slightly cramped using MDN's font, but according to many style guides, you are not supposed to put spaces anyway ^^

**Pull request merged**